src/containers/Cluster/Cluster.tsx

```diff
@@ -1,59 +1,175 @@
+    
+    const infoLoading = (clusterLoading && !clusterWasLoaded) || (nodesLoading && !nodesWasLoaded);
+
+    useEffect(() => {
+        dispatch(getClusterNodes());
+    }, [dispatch]);
+
+    useAutofetcher(
+        () => dispatch(getClusterInfo(clusterName ? String(clusterName) : undefined)),
+        [dispatch, clusterName],
+        true,
+    );
+
+    useEffect(() => {
+        dispatch(
+            setHeader([
+                {
+                    text: Name || 'Cluster',
+                    link: getClusterPath(),
+                },
+            ]),
+        );
+    }, [dispatch, Name]);
+
+    const versionToColor = useMemo(() => {
+        if (additionalVersionsProps?.getVersionToColorMap) {
+            return additionalVersionsProps?.getVersionToColorMap();
+        }
+        return parseVersionsToVersionToColorMap(cluster?.Versions);
+    }, [additionalVersionsProps, cluster]);
+
+    const versionsValues = useMemo(() => {
+        return parseNodesToVersionsValues(nodes, versionToColor);
+    }, [nodes, versionToColor]);
+
+    const renderTab = () => {
         switch (activeTab) {
-            case CLUSTER_PAGES.tenants.id: {
+            case clusterTabsIds.tenants: {
                 return <Tenants additionalTenantsInfo={additionalTenantsInfo} />;
             }
-            case CLUSTER_PAGES.nodes.id: {
+            case clusterTabsIds.nodes: {
                 return <Nodes additionalNodesInfo={additionalNodesInfo} />;
             }
-            case CLUSTER_PAGES.storage.id: {
+            case clusterTabsIds.storage: {
                 return <Storage additionalNodesInfo={additionalNodesInfo} />;
             }
-            case CLUSTER_PAGES.cluster.id: {
-                return (
-                    <ClusterInfo
-                        additionalClusterProps={additionalClusterProps}
-                        additionalVersionsProps={additionalVersionsProps}
-                    />
-                );
+            case clusterTabsIds.versions: {
+                return <Versions versionToColor={versionToColor} />;
+            }
+            default: {
+                return null;
+            }
+        }
+    };
+
+    const getTabEntityCount = (tabId: ClusterTab) => {
+        switch (tabId) {
+            case clusterTabsIds.tenants: {
+                return cluster?.Tenants;
+            }
+            case clusterTabsIds.nodes: {
+                return cluster?.NodesTotal;
             }
             default: {
                 return null;
             }
         }
     };
 
-    return <div className={b({tab: activeTab})}>{renderRoutes()}</div>;
+    return (
+        <div className={b()}>
+            <ClusterInfo
+                cluster={cluster}
+                versionsValues={versionsValues}
+                loading={infoLoading}
+                error={clusterError}
+                additionalClusterProps={additionalClusterProps}
+            />
+
+            <div>
```

Comment:
What do you think about moving tabs into separate component?

Reply:
I tried and not very liked it. It doesn't help much, but become complicated when we need to customise it (e.g. in Diagnostics there is autorefresh switch that is added to `Tabs`). So, in my view, it doesn't worth it